File: java-core/src/main/java/idesyde/core/headers/LabelledArcWithPorts.java
package idesyde.core.headers;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@JsonSerialize
public record LabelledArcWithPorts(
        String src,
        String dst,
        @JsonProperty("src_port")
        String srcPort,
        @JsonProperty("dst_port")
        String dstPort,
        String Label
) {
}

File: kotlin-core/src/main/kotlin/idesyde/core/headers/LabelledArcWithPorts.kt
package idesyde.core.headers

import kotlinx.serialization.*
@Serializable
data class LabelledArcWithPorts (
        val dst: String,

        @SerialName("dst_port")
        val dstPort: String? = null,

        val label: String? = null,
        val src: String,

        @SerialName("src_port")
        val srcPort: String? = null
)
